sdk/go/common/workspace/plugins.go

```diff
@@ -739,9 +825,17 @@ func (reader *checksumReader) Close() error {
 func (source *checksumSource) Download(
 	version semver.Version, opSy string, arch string,
 	getHTTPResponse func(*http.Request) (io.ReadCloser, int64, error),
+) (io.ReadCloser, int64, error) {
+	return source.DownloadWithContext(context.Background(), version, opSy, arch, getHTTPResponse)
+}
+
+func (source *checksumSource) DownloadWithContext(
```

Comment:
`sdk/go/common` is a bit of a weird place, and is not really part of the public API. This is mostly used by us in the various language SDKs and in the tfbridge.

I think we're ok with breaking the API here, especially when the fix is to simply pass a context.